sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala

```diff
@@ -437,6 +437,13 @@ object GpuDeviceManager extends Logging {
             s"dropping pinned memory to ${ret / 1024 / 1024.0} MiB")
         ret
       }
+      val nonPinnedLimit = finalMemoryLimit - totalOverhead - pinnedLimit
+      logWarning(s"Off Heap Host Memory configured to be " +
+          s"${pinnedLimit / 1024 / 1024.0} MiB pinned, " +
+          s"${nonPinnedLimit / 1024 / 1024.0} MiB non-pinned, and " +
+          s"${totalOverhead / 1024 / 1024.0} MiB of untracked overhead.")
+      HostAlloc.initialize(nonPinnedLimit)
```

Comment:
It looks like this will reference `PinnedMemoryPool.getTotalPoolSizeBytes`, but we don't initialize `PinnedMemoryPool` until line 452 below.

Reply:
Thanks for catching this. It should be fixed now. Please take another look.